pkg/config/runtime/runtime_http.go

```diff
@@ -17,23 +18,32 @@ func (c *Configuration) GetRoutersByEntryPoints(ctx context.Context, entryPoints
 			continue
 		}
 
+		logger := log.FromContext(log.With(ctx, log.Str(log.RouterName, rtName)))
 		eps := rt.EntryPoints
 		if len(eps) == 0 {
+			logger.Debugf("No entrypoint defined for this router, using the defaults ones instead: %+v", entryPoints)
 			eps = entryPoints
 		}
+		entryPointsCount := 0
 		for _, entryPointName := range eps {
 			if !contains(entryPoints, entryPointName) {
-				log.FromContext(log.With(ctx, log.Str(log.EntryPointName, entryPointName))).
+				rt.AddError(fmt.Errorf("entryPoint %q doesn't exist", entryPointName), false)
+				logger.WithField(log.EntryPointName, entryPointName).
 					Errorf("entryPoint %q doesn't exist", entryPointName)
 				continue
 			}
 
+			entryPointsCount++
 			if _, ok := entryPointsRouters[entryPointName]; !ok {
 				entryPointsRouters[entryPointName] = make(map[string]*RouterInfo)
 			}
 
 			entryPointsRouters[entryPointName][rtName] = rt
 		}
+		if entryPointsCount == 0 {
+			rt.AddError(fmt.Errorf("no valid entryPoints for this router"), true)
```

Comment:
```suggestion
			rt.AddError(fmt.Errorf("no valid entryPoint for this router"), true)
```
The current wording is grammatically correct, so this is only an optional suggestion. Your call. If you change it here, you need to change it below and in the tests too of course.